libs/providers/config-cat/src/lib/config-cat-provider.ts

```diff
@@ -1,42 +1,88 @@
 import {
   EvaluationContext,
+  GeneralError,
   JsonValue,
+  OpenFeatureEventEmitter,
   ParseError,
   Provider,
+  ProviderEvents,
   ResolutionDetails,
   ResolutionReason,
   StandardResolutionReasons,
   TypeMismatchError,
 } from '@openfeature/js-sdk';
-import * as configcat from 'configcat-js';
-import { IConfigCatClient } from 'configcat-js';
-import { SettingValue } from 'configcat-common/lib/RolloutEvaluator';
+import { getClient, IConfigCatClient, IEvaluationDetails, SettingValue } from 'configcat-js';
 import { transformContext } from './context-transformer';
 
 export class ConfigCatProvider implements Provider {
-  private client: IConfigCatClient;
+  private readonly clientParameters: Parameters<typeof getClient>;
+  public readonly events = new OpenFeatureEventEmitter();
+  private client?: IConfigCatClient;
 
-  private constructor(client: IConfigCatClient) {
-    this.client = client;
+  public metadata = {
+    name: ConfigCatProvider.name,
+  };
+
+  constructor(...params: Parameters<typeof getClient>) {
+    this.clientParameters = params;
   }
 
-  public static create(...params: Parameters<typeof configcat.getClient>) {
-    return new ConfigCatProvider(configcat.getClient(...params));
+  public static create(...params: Parameters<typeof getClient>) {
+    return new ConfigCatProvider(...params);
   }
 
-  public static createFromClient(client: IConfigCatClient) {
-    return new ConfigCatProvider(client);
+  public async initialize(): Promise<void> {
+    return new Promise((resolve) => {
+      const originalParameters = this.clientParameters;
+      const options = originalParameters[2];
+
+      if (options) {
+        const oldSetupHooks = options.setupHooks;
+
+        options.setupHooks = (hooks) => {
+          oldSetupHooks?.(hooks);
+
+          // After resolving, once, we can simply emit events the next time
+          hooks.once('clientReady', () => {
+            hooks.on('clientReady', () => this.events.emit(ProviderEvents.Ready));
+            this.events.emit(ProviderEvents.Ready);
+            resolve();
+          });
+
+          hooks.on('configChanged', (projectConfig) =>
+            this.events.emit(ProviderEvents.ConfigurationChanged, { metadata: { ...projectConfig } })
+          );
+
+          hooks.on('clientError', (message: string, error) =>
+            this.events.emit(ProviderEvents.Error, {
+              message: message,
+              metadata: error,
+            })
+          );
+        };
+      }
+
+      this.client = getClient(...originalParameters);
```

Comment:
This will cause the promise returned in the initialization to hang indefinitely if `options` is falsy.

Reply:
Oh yes, sure it does🤦‍♂️
Did not pay enough attention, fixed it thanks @beeme1mr.